pkg/controller/secret/secret_controller.go

```diff
@@ -40,6 +42,30 @@ func newReconciler(mgr manager.Manager) reconcile.Reconciler {
 	return &ReconcileSecret{client: mgr.GetClient(), scheme: mgr.GetScheme()}
 }
 
+// mySecretPredicate filters out any events not related to our Secret.
+var mySecretPredicate = predicate.Funcs{
+	CreateFunc: func(e event.CreateEvent) bool { return passes(e.Object) },
+	DeleteFunc: func(e event.DeleteEvent) bool { return passes(e.Object) },
+	// UpdateFunc passes if *either* the new or old object is one we care about.
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		return passes(e.ObjectOld) || passes(e.ObjectNew)
+	},
+	GenericFunc: func(e event.GenericEvent) bool { return passes(e.Object) },
+}
+
+func passes(o runtime.Object) bool {
+	if o == nil {
```

Comment:
Can this path actually be hit?

Reply:
It's what other examples of predicate funcs do, including [those in the predicate lib](https://github.com/kubernetes-sigs/controller-runtime/blob/fb1f0d625e8b902ac8388b2ef31acab9e7d2f19f/pkg/predicate/predicate.go#L117-L182) itself.

Put it this way: if `nil` is impossible, no harm no foul. But if it _is_ possible and we leave this check out... Watch() will `panic`.